### PR TITLE
HPC: Add test for slurm accounting

### PR DESCRIPTION
**Pull request by schlad**
- Verification run: http://10.160.65.14/tests/7954

### CODE REVIEW

File: tests/hpc/slurm_master_db.pm [hunk 1]
```diff
@@ -137,7 +137,73 @@ sub test_06 {
...
+    ##Add users TODO: surely this should be abstracted
+    script_run("useradd $users{user_1}");
+    script_run("useradd $users{user_2}");
+    script_run("useradd $users{user_3}");
+    script_run("useradd $users{user_4}");
+
+    script_run('sacctmgr -i add cluster linux');
+    my $cluster = script_output('sacctmgr -n -p list cluster');
+
+    if (index($cluster, 'linux') == -1) {
```
Comment (jlausuch) [inline tests/hpc/slurm_master_db.pm]: the output of the cmd is

linux||0|0|1||||||||normal||

maybe you want to make sure string "linux" is there? You could simple do something like:

    if ( $cluster !=~ m/linux/ ) {

File: tests/hpc/slurm_master_db.pm [hunk 2]
```diff
@@ -137,7 +137,73 @@ sub test_06 {
...
+    ### Create accounts in org=UNI_Y
+    script_run("sacctmgr -i add account UNI_Y_Physics Description=\"UNI_Y\" Organization=UNI_Y");
+    script_run("sacctmgr -i add account UNI_Y_Biology Description=\"UNI_Y\" Organization=UNI_Y");
+    #Add users associated with account in org=UNI_Y
+    script_run("sacctmgr -i create user name=Joe DefaultAccount=UNI_Y_Physics");
+    script_run("sacctmgr -i create user name=Noah DefaultAccount=UNI_Y_Biology");
+    script_run("sacctmgr -i create user name=$users{user_4} DefaultAccount=UNI_Y_Physics");
+    script_run("sacctmgr -i create user name=$users{user_3} DefaultAccount=UNI_Y_Biology");
+
+    script_run('sacctmgr show account');
```
Comment (jlausuch) [inline tests/hpc/slurm_master_db.pm]: what about record_info to get this information quicker without opening the log?

File: tests/hpc/slurm_master_db.pm (hunk 1 above)
Comment (schlad) [reply tests/hpc/slurm_master_db.pm]: sure, might be indeed more easy to read

File: tests/hpc/slurm_master_db.pm [hunk 3]
```diff
@@ -137,7 +137,73 @@ sub test_06 {
...
+    my $jobs = script_output("sacct -n -p --starttime 2010-01-01 --format=User,Account,JobID,Jobname,partition,state,time,start,end,elapsed,MaxRss,MaxVMSize,nnodes,ncpus,nodelist");
+
+    #check if there are expected srun jobs being recorded in the accounting db
+    $result = 1 unless (($jobs =~ /$users{user_1}/) &&
+        ($jobs =~ /$users{user_2}/) &&
+        ($jobs =~ /$users{user_3}/) &&
+        ($jobs =~ /$users{user_4}/));
+
+  FAIL:
+    my %results = generate_results($name, $description, $result);
```
Comment (jlausuch) [inline tests/hpc/slurm_master_db.pm]: you can directly pass 1 to generate_results and skip doing $result = 1 in line 161.

File: tests/hpc/slurm_master_db.pm (hunk 2 above)
Comment (schlad) [reply tests/hpc/slurm_master_db.pm]: make sense

File: lib/hpcbase.pm [hunk 4]
```diff
@@ -115,6 +115,8 @@ sed -i "/^NodeName.*/c\\NodeName=$cluster_ctl_nodes,$cluster_compute_nodes Socke
 sed -i "/^PartitionName.*/c\\PartitionName=normal Nodes=$cluster_ctl_nodes,$cluster_compute_nodes Default=YES MaxTime=24:00:00 State=UP" /etc/slurm/slurm.conf
```
Comment (asmorodskyi) [inline lib/hpcbase.pm]: some time ago I introduced this function :
https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/utils.pm#L1547 , might worth looking for this case . ( just a side note not related to this PR)

**Pull request merged**